File: src/main/java/org/eln2/sim/thermal/process/ResistorHeatThermalLoad.java
package org.eln2.sim.thermal.process;

/*
NOTE: DO NOT IMPORT MINECRAFT CODE IN THIS CLASS
EXTEND IT INSTEAD IN THE org.eln.nbt DIRECTORY
 */

import org.eln2.sim.process.IProcess;
import org.eln2.sim.thermal.ThermalLoad;
import org.eln2.sim.mna.component.Resistor;

public class ResistorHeatThermalLoad implements IProcess {

    Resistor r;
    ThermalLoad load;

    public ResistorHeatThermalLoad(Resistor r, ThermalLoad load) {
        this.r = r;
        this.load = load;
    }

    @Override
    public void process(double time) {
        load.movePowerTo(r.getP());
    }
}

File: src/main/java/org/eln2/sim/thermal/process/ResistorHeatThermalLoad.kt
package org.eln2.sim.thermal.process

/*
NOTE: DO NOT IMPORT MINECRAFT CODE IN THIS CLASS
EXTEND IT INSTEAD IN THE org.eln.nbt DIRECTORY
 */

import org.eln2.sim.mna.component.Resistor
import org.eln2.sim.process.IProcess
import org.eln2.sim.thermal.ThermalLoad

class ResistorHeatThermalLoad(var r: Resistor, var load: ThermalLoad) : IProcess {
    override fun process(time: Double) {
        load.movePowerTo(r.getP())
    }
}